packages/client/src/base/priority.ts

```diff
@@ -0,0 +1,50 @@
+/********************************************************************************
+ * Copyright (c) 2024 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { AnyObject, hasNumberProp } from '@eclipse-glsp/sprotty';
+
+/**
+ * A common interface for objects that can be orderable by a priority.
+ */
+export interface Prioritized {
+    priority: number;
+}
+
+export namespace Prioritized {
```

Comment:
We already have a `Ranked` API that more or less servers the same purpose.
Maybe it would be possible to combine those two APIS or at least deprecate on of them so that we don`t have 
two APIS for the same purpose in parallel?

Reply:
Yeah, I saw that and I wasn't entirely sure how to best go forward with this. They also seem to have reverted semantics, i.e., a lower rank means you are used before elements with a higher rank but for priority it is kind of the reverse, a high priority means you are before other elements. Sure, semantically they are the same but it was hard to choose which to deprecate if at all.

Reply:
I'll have another look at this tomorrow.

Reply:
I now use `rank` everywhere and introduced a backwards compatible property.